packages/core/contracts/cross-chain-oracle/GovernorSpoke.sol

```diff
@@ -37,15 +48,27 @@ contract GovernorSpoke is Lockable, ChildMessengerConsumerInterface {
      */
     function processMessageFromParent(bytes memory data) public override nonReentrant() onlyMessenger() {
         (address to, bytes memory inputData) = abi.decode(data, (address, bytes));
-        // TODO: Consider calling this via <address>.call(): https://docs.soliditylang.org/en/v0.8.10/units-and-global-variables.html?highlight=low%20level%20call#members-of-address-types
-        // to avoid inline assembly.
 
         require(_executeCall(to, inputData), "execute call failed");
-        emit ExecutedGovernanceTransaction(to, inputData);
+        emit ExecutedGovernanceTransaction(to, data);
+    }
+
+    /**
+     * @notice Changes the address of the child messenger contract used by both this contract and the OracleSpoke.
+     * @dev Can only be called by this contract. i.e Executed by the GovernorHub sending a message, via the messenger
+     * to the processMessageFromParent function which intern calls back into this contract.
+     * @param newChildMessenger Address of the child messenger contract to set for this contract and OracleSpoke.
+     */
+    function setChildMessenger(address newChildMessenger) public onlyThisContract {
```

Comment:
Nit: should this be reentrant guarded?

Reply:
the method reverts if this has a reentrant guard on it :)